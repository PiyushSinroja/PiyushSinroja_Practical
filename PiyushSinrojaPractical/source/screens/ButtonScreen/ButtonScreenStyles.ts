import {StyleSheet, Dimensions} from 'react-native';
import {globalColors} from '../../constant/Colors';
const {height} = Dimensions.get('screen');

/**
 * Button Screen Styles
 */
export const ButtonScreenStyles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    flex: 1,
    justifyContent: 'space-between',
  },
  safeAreaStyle: {
    backgroundColor: 'black',
    flex: 1,
  },
  userDetailView: {
    height: 40,
    justifyContent: 'flex-end',
    flexDirection: 'row',
  },
  buttonsContainer: {
    justifyContent: 'flex-end',
  },
  textFourVariations: {
    fontWeight: 'normal',
    color: '#B1BD2F',
    fontSize: 19,
  },
  viewFourVariations: {
    flexDirection: 'column',
    width: '100%',
    marginBottom: 10,
    height: height * 0.06,
    justifyContent: 'center',
    alignItems: 'center',
  },
  sliderButtonContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    marginStart: 20,
    marginEnd: 20,
    height: height * 0.06,
  },
  swipeButtonParentViewContainer: {
    flexDirection: 'column',
    width: '100%',
    marginBottom: 20,
  },
  text: {
    color: globalColors.white,
    fontSize: 19,
    paddingVertical: 10,
    textAlign: 'left',
  },
  userText: {
    marginLeft: 10,
    marginRight: 15,
  },
  buttonTextStyle: {
    color: '#65ABE3',
  },
  firstButtonStyle: {
    backgroundColor: globalColors.black,
  },
  secondButtonStyle: {
    backgroundColor: '#2E3A41',
  },
  thirdButtonStyle: {
    backgroundColor: globalColors.sky,
  },
});
