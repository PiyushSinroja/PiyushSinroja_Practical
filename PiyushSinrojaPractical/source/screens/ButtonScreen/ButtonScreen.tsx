import React, { useState, ReactElement} from 'react';
import {SafeAreaView, View, Text, Image} from 'react-native';
import {ButtonScreenStyles} from './ButtonScreenStyles';
import CommonButton from '../../common/components/CommonButton';
import {globalColors} from '../../constant/Colors';
import SwipeButton from 'rn-swipe-button';
import {useSelector} from 'react-redux';

const diamondIcon = require('../../assets/images/diamondIcon.png');

interface ButtonScreenProps {
  userFullName?: string;
  navigation?: any;
}

/**
 * button screen used to display button variations and user information
 *
 * @param {*} {navigation}
 * @returns jsx
 */
const ButtonScreen: React.FC<ButtonScreenProps> = ({navigation}) => {
  const [isUnlock, setIsUnlock] = useState(false);

  const userFullName = useSelector((state: any) => state.UserReducers.userName);

  const navigateToInformationScreen = () => {
    navigation?.navigate('InformationScreen');
  };

  const sliderIcon = () => <Image resizeMode="center" source={diamondIcon} />;

  const viewSliderButton = () => {
    let titleSlider = isUnlock == false ? 'Slide to continue' : 'Success';

    return (
      <View style={ButtonScreenStyles.swipeButtonParentViewContainer}>
        <SwipeButton
          containerStyles={ButtonScreenStyles.sliderButtonContainer}
          thumbIconBackgroundColor={globalColors.sky}
          thumbIconBorderColor={globalColors.transparent}
          thumbIconComponent={sliderIcon}
          thumbIconStyles={{
            borderRadius: 5,
          }}
          railBorderColor={globalColors.grey}
          railBackgroundColor={globalColors.black}
          onSwipeStart={() => {
            setIsUnlock(false);
          }}
          onSwipeSuccess={() => {
            setIsUnlock(true);
          }}
          shouldResetAfterSuccess={false}
          title={titleSlider}
          titleColor={globalColors.sky}
          railStyles={{
            backgroundColor: globalColors.transparent,
            borderColor: globalColors.transparent,
          }}
        />
      </View>
    );
  };

  return (
    <SafeAreaView style={ButtonScreenStyles.safeAreaStyle}>
      <View style={ButtonScreenStyles.container}>
        <View style={ButtonScreenStyles.userDetailView}>
          <Text style={ButtonScreenStyles.text}> User Name:</Text>
          <Text style={[ButtonScreenStyles.text, ButtonScreenStyles.userText]}>
            {userFullName}
          </Text>
        </View>
        <View style={ButtonScreenStyles.buttonsContainer}>
          <View style={ButtonScreenStyles.viewFourVariations}>
            <Text style={ButtonScreenStyles.textFourVariations}>
              4 variations of a button
            </Text>
          </View>

          <CommonButton
            onPress={() => {
              navigateToInformationScreen();
            }}
            titleString={'Press me'}
            style={ButtonScreenStyles.firstButtonStyle}
            textStyle={ButtonScreenStyles.buttonTextStyle}
          />

          <CommonButton
            onPress={() => {
              navigateToInformationScreen();
            }}
            titleString={'Press me'}
            style={ButtonScreenStyles.secondButtonStyle}
            textStyle={ButtonScreenStyles.buttonTextStyle}
          />

          <CommonButton
            onPress={() => {
              navigateToInformationScreen();
            }}
            titleString={'Press me'}
            style={ButtonScreenStyles.thirdButtonStyle}
          />

          {viewSliderButton()}
        </View>
      </View>
    </SafeAreaView>
  );
};

export default ButtonScreen;
