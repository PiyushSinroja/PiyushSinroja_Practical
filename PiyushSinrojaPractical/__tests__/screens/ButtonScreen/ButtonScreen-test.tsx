import React from 'react';
import ButtonScreen from '../../../source/screens/ButtonScreen/ButtonScreen';
import {render, fireEvent} from '@testing-library/react-native';

import {Provider} from 'react-redux';
import configureStore from '../../../source/redux-classes/ConfigStore';

const store = configureStore();

import SwipeButton from 'rn-swipe-button';

describe('Button Screen Test', () => {
  it('renders properly', () => {
    const {toJSON, UNSAFE_getByType} = render(
      <Provider store={store}>
        <ButtonScreen />
      </Provider>,
    );

    const sliderButton = UNSAFE_getByType(SwipeButton);
    fireEvent(sliderButton, 'onSwipeStart');
    fireEvent(sliderButton, 'onSwipeSuccess');
    expect(toJSON()).toMatchSnapshot();
  });
});
