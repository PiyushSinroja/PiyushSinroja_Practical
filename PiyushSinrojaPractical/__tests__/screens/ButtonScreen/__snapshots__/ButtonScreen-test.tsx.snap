// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Screen Test renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "paddingVertical": 10,
            "textAlign": "left",
          }
        }
      >
         User Name:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 19,
              "paddingVertical": 10,
              "textAlign": "left",
            },
            Object {
              "marginLeft": 10,
              "marginRight": 15,
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": 80.03999999999999,
            "justifyContent": "center",
            "marginBottom": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B1BD2F",
              "fontSize": 19,
              "fontWeight": "normal",
            }
          }
        >
          4 variations of a button
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "black",
              "borderRadius": 5,
              "height": 80.03999999999999,
              "justifyContent": "center",
              "marginEnd": 20,
              "marginStart": 20,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 17,
                  "fontWeight": "normal",
                },
                Object {
                  "color": "#65ABE3",
                },
              ]
            }
          >
            Press me
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2E3A41",
              "borderRadius": 5,
              "height": 80.03999999999999,
              "justifyContent": "center",
              "marginEnd": 20,
              "marginStart": 20,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 17,
                  "fontWeight": "normal",
                },
                Object {
                  "color": "#65ABE3",
                },
              ]
            }
          >
            Press me
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5AAAF2",
              "borderRadius": 5,
              "height": 80.03999999999999,
              "justifyContent": "center",
              "marginEnd": 20,
              "marginStart": 20,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 17,
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }
          >
            Press me
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 50,
                "borderWidth": 1,
                "justifyContent": "center",
                "margin": 5,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "grey",
                "borderRadius": 5,
                "height": 80.03999999999999,
                "justifyContent": "center",
                "marginEnd": 20,
                "marginStart": 20,
              },
            ]
          }
        >
          <Text
            importantForAccessibility=""
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "position": "absolute",
                },
                Object {
                  "color": "#5AAAF2",
                  "fontSize": 20,
                },
              ]
            }
          >
            Success
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
